apollo-runtime/src/test/java/com/apollographql/android/impl/InterceptorTest.java

```diff
@@ -0,0 +1,65 @@
+package com.apollographql.android.impl;
+
+import com.apollographql.android.api.graphql.Query;
+import com.apollographql.android.api.graphql.ResponseFieldMapper;
+import com.apollographql.android.api.graphql.ResponseReader;
+import com.apollographql.android.impl.util.HttpException;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import okhttp3.OkHttpClient;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+import static com.google.common.truth.Truth.assertThat;
+import static junit.framework.Assert.fail;
+
+public class InterceptorTest {
+  private ApolloClient apolloClient;
+  @Rule public final MockWebServer server = new MockWebServer();
+
+  @Before public void setUp() {
+    apolloClient = ApolloClient.builder()
+        .serverUrl(server.url("/"))
+        .okHttpClient(new OkHttpClient.Builder().build())
+        .build();
+  }
+
+  @Test public void httpException() throws Exception {
+    server.enqueue(new MockResponse().setResponseCode(401).setBody("Unauthorized request!"));
+
+    try {
+      apolloClient.newCall(new Query() {
+        @Override public String queryDocument() {
+          return "";
+        }
+
+        @Override public Variables variables() {
+          return EMPTY_VARIABLES;
+        }
+
+        @Override public ResponseFieldMapper<Data> responseFieldMapper() {
+          return new ResponseFieldMapper<Data>() {
+            @Override public Data map(ResponseReader responseReader) throws IOException {
+              return null;
+            }
+          };
+        }
+
+        @Override public Object wrapData(Data data) {
+          return data;
+        }
+      }).execute();
+      fail("Expected HttpException");
+    } catch (HttpException httpException) {
+      assertThat(httpException.code()).isEqualTo(401);
+      assertThat(httpException.message()).isEqualTo("Client Error");
+      assertThat(httpException.rawResponse().body().string()).isEqualTo("Unauthorized request!");
```

Comment:
Is is a good idea to not close the body of the response? I am not sure of the implications of not doing so. Personally, I would have changed the `HttpException` class to have a static method that looks like:

```java
  public static HttpException from(Response response) throws IOException {
     try {
       String body = response.body() != null ? response.body().string() : "";
       return new HttpException(response, response.code(), response.message(), body);
     } finally {
       closeQuietly(response);
     }
   }
```

So that the body does get closed but before it does, we have read the contents of it. Also IIRC, the `ResponseBody` in the `Response` object is what gets closed when `close` is called. So if we read and copy the contents over, it is probably safe to close.